server_update.py

```diff
@@ -340,7 +340,7 @@ def __init__(self, path):
 
         self.path = path  # Path to file being updated
         self.temp = None  # Tempdir instance
-        self._working_path = os.path.join(os.path.dirname(__file__), 'server_update')  # Working directory path
+        self._working_path = os.path.join(os.path.dirname(self.path), 'server_update')  # Working directory path

```

Comment:
Your code is specifying the .jar file for the working directory, should instead be the directory the .jar file is located in. This can be achieved using the pathlib module.

Reply:
`os.path.dirname` will get the path of the `.jar` file so I think this should be fine

Reply:
Yes, good point, did not see that. I will do some testing and merge your request.